mono/metadata/sgen-mono.c

```diff
@@ -2162,37 +2166,38 @@ sgen_client_scan_thread_data (void *start_nursery, void *end_nursery, gboolean p
 	return;
 #endif
 
-	FOREACH_THREAD_EXCLUDE (info, MONO_THREAD_INFO_FLAGS_NO_GC) {
+	FOREACH_THREAD_EXCLUDE (mono_thread_info, MONO_THREAD_INFO_FLAGS_NO_GC) {
+		SgenThreadInfo *info = (SgenThreadInfo*)mono_thread_info;
```

Comment:
IDK about having two variables aliasing to the same value.
I think we should stick with a single one and emit casts in the very few places that needs it.
As you mentioned on slack, moving stack_start would fix most of this.

Reply:
The problem with moving stack_start is I have to touch all the uses. ok?
Also could keep FOREACH factored on type, that might help. I can try.